Jenkinsfile

```diff
@@ -470,355 +453,395 @@ def testGo(prefix, packagesToTest) {
-              ]) {
-                println "Running tests for ${testSpec.dirPath}"
-                def t = getOverallTimeout(testSpec)
-                timeout(activity: true, time: t.time, unit: t.unit) {
-                  sh "citogo --flakes 3 --fails 3 --build-id ${env.BUILD_ID} --branch ${env.BRANCH_NAME} --prefix ${testSpec.dirPath} --s3bucket ci-fail-logs --build-url ${env.BUILD_URL} --no-compile --test-binary ./${testBinary} --timeout 150s ${testSpec.citogo_extra || ''}"
-                }
-              }
+          withCredentials([
+            [$class: 'StringBinding', credentialsId: 'citogo-flake-webhook', variable : 'CITOGO_FLAKE_WEBHOOK'],
+            [$class: 'StringBinding', credentialsId: 'citogo-aws-secret-access-key', variable : 'CITOGO_AWS_SECRET_ACCESS_KEY'],
+            [$class: 'StringBinding', credentialsId: 'citogo-aws-access-key-id', variable : 'CITOGO_AWS_ACCESS_KEY_ID'],
+            [$class: 'StringBinding', credentialsId: 'citogo-master-fail-webhook', variable : 'CITOGO_MASTER_FAIL_WEBHOOK']
+          ]) {
+            println "Running tests for ${testSpec.dirPath}"
+            def t = getOverallTimeout(testSpec)
+            timeout(activity: true, time: t.time, unit: t.unit) {
+              sh "citogo --flakes 3 --fails 3 --build-id ${env.BUILD_ID} --branch ${env.BRANCH_NAME} --prefix ${testSpec.dirPath} --s3bucket ci-fail-logs --build-url ${env.BUILD_URL} --no-compile --test-binary ./${testBinary} --timeout 150s ${testSpec.citogo_extra || ''}"
             }
           }
-          if (testSpec.name in specialTestFilter) {
-            specialTests["${prefix}${testSpec.name}"] = test
-          } else {
-            tests["${prefix}${testSpec.name}"] = test
-          }
-          println "Will run tests for ${testSpec.dirPath}"
-        } else {
-          println "Skipping tests for ${testSpec.dirPath} because no test binary was produced."
         }
       }
     }
-
-    // Schedule the tests
-    def parallelTests = []
-    def testBatch = [:]
-    tests.each { name, closure ->
-      if (testBatch.size() == 4) {
-        parallelTests << testBatch
-        testBatch = [:]
-      }
-      testBatch[name] = closure
+    if (testSpec.name in specialTestFilter) {
+      specialTests["${prefix}${testSpec.name}"] = test
+    } else {
+      tests["${prefix}${testSpec.name}"] = test
     }
-    if (testBatch.size() > 0) {
+  }
+  }
+
+  stage("Run Test Suite") {
+  // Schedule the tests
+  def parallelTests = []
+  def testBatch = [:]
+  tests.each { name, closure ->
+    if (testBatch.size() == 6) {
       parallelTests << testBatch
+      testBatch = [:]
     }
-    parallelTests << specialTests
-    helpers.waitForURL(prefix, env.KEYBASE_SERVER_URI)
-    parallelTests.each { batch ->
-      parallel(batch)
-    }
-  }}
+    testBatch[name] = closure
+  }
+  if (testBatch.size() > 0) {
+    parallelTests << testBatch
+  }
+  parallelTests << specialTests
+  waitForURL(prefix, env.KEYBASE_SERVER_URI)
```

Comment:
Why did you copy these helpers into here?

Reply:
So I think what happened is, I got rid of a lot of the upfront checking before the test suite started running, which means that the server has less time to start up (before it was doing that while linting and whatever was going on). So I needed to extend timeout or it was flaky.

Should I parametrize the timeout value and add a helpers function?

Reply:
Yeah, please parameterize it (which you already did in keybase/jenkins-helpers#12).